### PR TITLE
PLT-5140: Display loading screen when switching team

### LINKED ISSUE #4997: [Help Wanted] Add a loading screen while switching teams via the teams sidebar
If you're interested please comment here and come [join our "Contributors" community channel](https://pre-release.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://pre-release.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://docs.mattermost.com/guides/developer.html), specifically for [machine setup](https://docs.mattermost.com/developer/developer-setup.html) and for [developer workflow](https://docs.mattermost.com/developer/developer-flow.html).

----

**Notes**: [Jira ticket](https://mattermost.atlassian.net/browse/PLT-5140)

The same loading screen that appears when you open up a team, should appear while switching to another team as the center channel and left-hand sidebar loads.

See attached example

image

### PR DESCRIPTION
**Pull request by mikaoelitiana**
#### Summary
Display loading screen when switching team and waiting data to load

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-5140

#### Checklist
- [x] Has UI changes

Fixes #4997

### CODE REVIEW

**Comment by jasonblais** [comment]
Thanks @mikaoelitiana, appreciate your help here.

Wondering if we'd want to incorporate the changes you made in https://github.com/mattermost/platform/pull/5383 to this PR as well? It might be good to see how it all fits together.

**Comment by mikaoelitiana** [comment]
@jasonblais I will merge the other branch here.

**Comment by jasonblais** [comment]
Thanks @mikaoelitiana, appreciate it. Did you want to update anything else before I review the PR?

**Comment by mikaoelitiana** [comment]
@jasonblais no it's all there I think

File: config/config.json [hunk 1]
```diff
@@ -41,6 +41,7 @@
         "PostEditTimeLimit": 300,
         "TimeBetweenUserTypingUpdatesMilliseconds": 5000,
         "EnableUserTypingMessages": true,
+        "EnableUserTypingMessages": true,
```
Comment (jasonblais) [inline config/config.json]: @mikaoelitiana can you help remove this change in config.json?

Our fault, we originally had this setting duplicated in config.json which got fixed in `master`.

**Comment by mikaoelitiana** [comment]
@jasonblais I will remove that quickly

**Comment by jasonblais** [comment]
Perfect, thank you @mikaoelitiana!

**Pull request merged**